### PR TITLE
WIP [helm] Add loki and promtail

### PR DESCRIPTION
**Pull request by aledbf**
Test loki and promtail to forward logs from workspace pod logs and stream it from a loki query using minio as object-store.

Test:

```
kubectl port-forward loki-0 3100 &
```

```
curl -vvv -G -s  "http://localhost:3100/loki/api/v1/query" --data-urlencode 'query=sum(rate({component="controller"}[10m])) by (level)'
*   Trying ::1:3100...
* TCP_NODELAY set
* Connected to localhost (::1) port 3100 (#0)
> GET /loki/api/v1/query?query=sum%28rate%28%7Bcomponent%3D%22controller%22%7D%5B10m%5D%29%29%20by%20%28level%29 HTTP/1.1
> Host: localhost:3100
> User-Agent: curl/7.68.0
> Accept: */*
> 
Handling connection for 3100
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Type: application/json; charset=UTF-8
< Date: Wed, 03 Mar 2021 22:03:43 GMT
< Content-Length: 701
< 
{"status":"success","data":{"resultType":"vector","result":[{"metric":{},"value":[1614809023.14,"0.03833333333333333"]}],"stats":{"summary":{"bytesProcessedPerSecond":4982503,"linesProcessedPerSecond":12867,"totalBytesProcessed":8906,"totalLinesProcessed":23,"execTime":0.001787455},"store":{"totalChunksRef":0,"totalChunksDownloaded":0,"chunksDownloadTime":0,"headChunkBytes":0,"headChunkLines":0,"decompressedBytes":0,"decompressedLines":0,"compressedBytes":0,"totalDuplicates":0},"ingester":{"totalReached":1,"totalChunksMatched":2,"totalBatches":0,"totalLinesSent":0,"headChunkBytes":8906,"headChunkLines":23,"decompressedBytes":0,"decompressedLines":0,"compressedBytes":0,"totalDuplicates":0}}}}
* Connection #0 to host localhost left intact
```

**TODO:**
- [x] Create a bucket in minio
- [x] Refactor chart parameters to avoid duplication
- [x] Remove use of client-go `GetLogs`
- [ ] Stream pod logs from loki
- [X] Logs per workspace

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -530,3 +527,121 @@ mysql:
...
+          pipeline_stages:
+            {{- toYaml .Values.config.snippets.pipelineStages | nindent 4 }}
+          kubernetes_sd_configs:
+            - role: pod
+          relabel_configs:
+            # Drop everything that is not a gitpod workspace
+            - source_labels:
+              - __meta_kubernetes_pod_label_app
+              - __meta_kubernetes_pod_label_component
+              regex: workspace
```
Comment (ArthurSens) [inline chart/values.yaml:562]: I think that the way it is now wouldn't work, it is trying to match the labels `app` and `component` and their values combined should be `workspace`.


We would need to choose between one of the two 👇 

```suggestion
              - __meta_kubernetes_pod_label_app
              - __meta_kubernetes_pod_label_component
              regex: gitpod;workspace
```
or

```suggestion
              - __meta_kubernetes_pod_label_component
              regex: gitpod;workspace
```

**Comment by aledbf** [comment]
Stream log example:
```
curl \
    --get \
    --include \
    --no-buffer \
    --header "Connection: Upgrade" \
    --header "Upgrade: websocket" \
    --header "Sec-WebSocket-Version: 13" \
    --header "Sec-WebSocket-Key: qwerty" \
    'http://localhost:3100/loki/api/v1/tail' \
    --data-urlencode 'query={workspace="ws-3b4a3ca6-8428-470b-b325-7706f5f0a288"}' --output -
```

**Pull request closed without merging**